payment-request/throw_if_not_active.https.html

```diff
@@ -0,0 +1,166 @@
+    "/payment-request/resources/page1.html"
+  );
+
+  // We navigate the iframe again, putting request1's document into an inactive state.
+  const request2 = await getLoadedPaymentRequest(
+    iframe,
+    "/payment-request/resources/page2.html"
+  );
+
+  // Now, request1's relevant global object's document is no longer active.
+  // So, call .show(), and make sure it rejects appropriately.
+  await promise_rejects(
+    t,
+    "AbortError",
+    request1.show(),
+    "Inactive document, so must throw AbortError"
+  );
+
+  // request2 has an active document tho, so confirm it's working as expected:
+  request2.show();
+  await request2.abort();
+  await promise_rejects(
+    t,
+    "InvalidStateError",
+    request2.show(),
+    "Abort already called, so InvalidStateError"
+  );
+  // We are done, so clean up.
+  iframe.remove();
+}, "PaymentRequest.show() aborts if the document is not active");
+
+promise_test(async t => {
+  // check that PaymentRequests can be constructed.
+  new PaymentRequest(validMethods, validDetails);
+  // We create two iframes, and nest them. Wait for both of them to load.
+  const iframe = document.createElement("iframe");
+  iframe.allowPaymentRequest = true;
+  document.body.appendChild(iframe);
+  // We get the request from the inner-most iframe.
+  const request = await new Promise(resolve => {
+    iframe.addEventListener(
+      "load",
+      () => {
+        const innerIframe = iframe.contentDocument.createElement("iframe");
+        innerIframe.allowPaymentRequest = true;
+        innerIframe.addEventListener(
+          "load",
+          () => {
+            // We create an instance of the inner iframe's PaymentRequest()
+            const { PaymentRequest } = innerIframe.contentWindow;
+            const request = new PaymentRequest(validMethods, validDetails);
+            resolve(request);
+          },
+          { once: true }
+        );
+        innerIframe.src = "/payment-request/resources/page2.html";
+        iframe.contentDocument.body.appendChild(innerIframe);
+      },
+      { once: true }
+    );
+    iframe.src = "/payment-request/resources/page1.html";
+  });
+
+  // Navigate the outer iframe to a new location. Wait for the load event to fire.
+  await new Promise(resolve => {
+    iframe.addEventListener("load", resolve);
+    iframe.src = "/payment-request/resources/page2.html";
+  });
+  // Now, request's relevant global object's document is still active
+  // (it is the active document of the inner iframe), but is not fully active
+  // (since the parent of the inner iframe is itself no longer active).
+  // So, call pr.show() and make sure it rejects appropriately.
+  await promise_rejects(
+    t,
+    "AbortError",
+    request.show(),
+    "Active, but not fully active, so must throw AbortError"
+  );
+  // We are done, so clean up.
+  iframe.remove();
+}, "PaymentRequest.show() aborts if the document is active, but not fully active");
+
+promise_test(async t => {
```

Comment:
This test is redundant with the first one, right?

Reply:
Um, maybe I screwed it up... In the first test, I'm testing that .show() can't be called if the document is inactive... but in this test, I'm making sure that that if the payment sheet is *already showing* then it should abort.

Reply:
Ah, no, you're totally right!!

I guess you can also use your helper in this test.

Reply:
Used it below ... was worried about adding too much indirection, but it's all good :)